File: 709.转换成小写字母.js
/*
 * @lc app=leetcode.cn id=709 lang=javascript
 *
 * [709] 转换成小写字母
 */

// @lc code=start
/**
 * @param {string} s
 * @return {string}
 */
var toLowerCase = function (s) {
  // 语法糖
  // return s.toLowerCase()

  // ascii码
  // 大写字母 ASCII 码: [65, 90]
  // 小写字母 ASCII 码: [97, 122]
  const sn = [] // 存放小写结果
  for (let ch of s) {
    if (ch.charCodeAt() >= 65 && ch.charCodeAt() <= 90) {
      // 按位运算 直接+32也可以
      ch = String.fromCharCode(ch.charCodeAt() | 32)
    }
    sn.push(ch)
  }
  return sn.join('')
}
// @lc code=end

File: 709.转换成小写字母.rs
/*
 * @lc app=leetcode.cn id=709 lang=rust
 *
 * [709] 转换成小写字母
 */

// @lc code=start
impl Solution {
    pub fn to_lower_case(s: String) -> String {
        // s.chars()
        //     .map(|c| {
        //         if c >= 'A' && c <= 'Z' {
        //             (c as u8 + 32) as char
        //         } else {
        //             c
        //         }
        //     })
        //     .collect()
        s.to_lowercase()
    }
}
// @lc code=end

fn main() {
    let s: &str = "Hello";
    let result = Solution::to_lower_case(s);
    println!("{}", result);
}